### PR TITLE
Expose contact point and contact normal on VehicleWheel3D to scripting.

### LINKED ISSUE #10106: Expose `m_raycastInfo` in VehicleWheel3D to GDScript.
### Describe the project you are working on

I'm working on a vehicular combat game with lots of verticality. The vehicles are in the air quite a bit, and the suspensions are taxed a lot.

### Describe the problem or limitation you are having in your project

The current implementation of the VehicleBody3D and VehicleWheel3D makes it impossible to collect the raycast information from the suspension. I have no good way to place the tire marks on the ground without being able to collect the collision point and normal from the wheels.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Exposing the raycast information that's already present in the wheel will make it easy to implement this feature, as I will know the collision point and normal that the wheel has detected.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Either a collection of methods like `get_contact_normal()` or `get_contact_position()`,  or simply exposing the entire `m_raycastInfo` struct to GDScript will work fine. The data is already there and maintained as seen in https://github.com/godotengine/godot/blob/master/scene/3d/physics/vehicle_body_3d.h. It should be trivial for a C++ engineer to go in and expose these.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

The current workaround is to create a second raycast for each of the wheels to re-do the work, or attempt to reverse the collision point (but crucially not the normal) from the wheel's rendered position. 

### Is there a reason why this should be core and not an add-on in the asset library?

I'm not aware of an asset library or addon that can reach into the C++ code and expose properties like this.

### PR DESCRIPTION
**Pull request by TestSubject06**
* *Bugsquad edit, closes: https://github.com/godotengine/godot-proposals/issues/10106*

### CODE REVIEW

**Comment by TestSubject06** [comment]
The style formatter threw up, and didn't really tell me why.

**Comment by AThousandShips** [comment]
Indentation, you can set up a pre-commit hook to help, see [here](https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html#pre-commit-hook)

**Comment by AThousandShips** [comment]
Is this meant to be a daft? Then please convert it to a draft, see top left of the page

**Comment by TestSubject06** [comment]
> Indentation

Okay, but what file is it even complaining about the indentation for? Everything is exactly like all surrounding code, including the documentation.

**Comment by TestSubject06** [comment]
Ah, I see. It didn't like that I had a space before the newline.

**Comment by TestSubject06** [comment]
@AThousandShips What do I need to do to get this moving? Is there someone I should ping to get this on a list for a review?

**Comment by AThousandShips** [comment]
We're in feature freeze until after we've released 4.3, until then focus is on bug fixes

**Comment by TestSubject06** [comment]
Now that 4.3 is out, can I request a review of this?

File: doc/classes/VehicleWheel3D.xml [hunk 1]
```diff
@@ -18,6 +18,20 @@
 				Returns [code]null[/code] if the wheel is not in contact with a surface, or the contact body is not a [PhysicsBody3D].
 			</description>
 		</method>
+		<method name="get_contact_normal" qualifiers="const">
+			<return type="Vector3" />
+			<description>
+				Returns the normal of the suspension's collision in world space if the wheel is in contact.
+				Otherwise, returns a vector pointing directly along the suspension axis toward the vehicle, in world space.
```
Comment (Calinou) [inline doc/classes/VehicleWheel3D.xml:25]: Merge into a single paragraph (for more compact description), and tweak the description to be more explicit:

```suggestion
				Returns the normal of the suspension's collision in world space if the wheel is in contact. If the wheel isn't in contact with anything, returns a vector pointing directly along the suspension axis toward the vehicle in world space.
```

File: doc/classes/VehicleWheel3D.xml [hunk 2]
```diff
@@ -18,6 +18,20 @@
...
+			</description>
+		</method>
+		<method name="get_contact_point" qualifiers="const">
+			<return type="Vector3" />
+			<description>
+				Returns the point of the suspension's collision in world space if the wheel is in contact.
+				Otherwise, returns the maximum point of the wheel's ray cast, in world space, defined by [code]wheel_rest_length + wheel_radius[/code].
```
Comment (Calinou) [inline doc/classes/VehicleWheel3D.xml:32]:
```suggestion
				Returns the point of the suspension's collision in world space if the wheel is in contact. If the wheel isn't in contactg with anything, returns the maximum point of the wheel's ray cast in world space, which is defined by [code]wheel_rest_length + wheel_radius[/code].
```

**Review by Calinou — APPROVED** [review]
Tested locally (rebased on top of `master` 568589c9d8c763bfb3a4348174d53b42d7c59f21), it works as expected.

**Testing project:** [truck_town_wheel_visualizer.zip](https://github.com/user-attachments/files/16720666/truck_town_wheel_visualizer.zip)

https://github.com/user-attachments/assets/cfd99bef-90fd-4b69-b7bf-fcba8c1dee6a

**Comment by TestSubject06** [comment]
Is this good to go with normal github squash+merge, or do I need to manually squash all the commits first?

**Comment by AThousandShips** [comment]
You need to squash the commits yourself, we do not use squash commits as GitHub currently do not allow combining squash *and* merge commits and we prefer having distinct merge commits as they make the history a lot easier to work with

File: doc/classes/VehicleWheel3D.xml [hunk 3]
```diff
@@ -18,6 +18,18 @@
...
+		<method name="get_contact_normal" qualifiers="const">
+			<return type="Vector3" />
+			<description>
+				Returns the normal of the suspension's collision in world space if the wheel is in contact. If the wheel isn't in contact with anything, returns a vector pointing directly along the suspension axis toward the vehicle in world space.
+			</description>
+		</method>
+		<method name="get_contact_point" qualifiers="const">
+			<return type="Vector3" />
+			<description>
+				Returns the point of the suspension's collision in world space if the wheel is in contact. If the wheel isn't in contactg with anything, returns the maximum point of the wheel's ray cast in world space, which is defined by [code]wheel_rest_length + wheel_radius[/code].
```
Comment (AThousandShips) [inline doc/classes/VehicleWheel3D.xml:30]:
```suggestion
				Returns the point of the suspension's collision in world space if the wheel is in contact. If the wheel isn't in contact with anything, returns the maximum point of the wheel's ray cast in world space, which is defined by [code]wheel_rest_length + wheel_radius[/code].
```

**Comment by TestSubject06** [comment]
Okay, everything's squashed up. Let me know if there's anything else I need to do here.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**